schematic/configuration.py

```diff
@@ -0,0 +1,98 @@
+import os
+import yaml
+
+
+class Configuration(object):
+
+    def __init__(self):
+        # this is the project's root directory
+        self.ROOT_DIR = None
+        # path to config.yml file
+        self.CONFIG_PATH = None
+        # path to credentials.json file
+        self.CREDS_PATH = None
+        # path to token.pickle file
+        self.TOKEN_PICKLE = None
+        # path to data directory
+        self.DATA_PATH = None
+        # path to service account credentials file
+        self.SERVICE_ACCT_CREDS = None
+        # path to synapse config file
+        self.SYNAPSE_CONFIG_PATH = None
+        # entire configuration data
+        self.DATA = None
+
+
+    def __getattribute__(self, name):
```

Comment:
This method essentially implements "lazy loading" of the configuration. Whenever the user or the module tries to access an attribute using the dot syntax (_e.g._ `schematic.TOKEN_PICKLE`), this method will check if the value is `None`, which means that nothing has been loaded so far. In most cases, @sujaypatil96 confirmed that in most cases, `schematic` will be used via one of the provided scripts, which should load a config file as demonstrated in `csv_to_schemaorg.py`. 

As these interfaces are implemented, I added a quick way of passing that argument without a CLI using the environment variable `SCHEMATIC_CONFIG`. The two commands below achieve the same thing, but you can see in the output of the second command, the config was pulled from the environment variable. 

```
$ python ~/repos/schematic/examples/csv_to_schemaorg.py --config ~/repos/schematic/config.yml ~/repos/schematic/data/csv/example.csv && echo "Exit code: $status"

[...]
Exit code: 0


$ SCHEMATIC_CONFIG=~/repos/schematic/config.yml python ~/repos/schematic/examples/csv_to_schemaorg.py ~/repos/schematic/data/csv/example.csv && echo "Exit code: $status"

Loading config YAML file specified in 'SCHEMATIC_CONFIG' environment variable: /Users/bgrande/repos/schematic/config.yml
[...]
Exit code: 0
```